### PR TITLE
Draft of modslot:any and modslot:none filters.

### PR DESCRIPTION
**Pull request by johyphenel**
Apologies in advance, I don't use github much, new to node and typescript, new to DIM. Please take a look at this first draft and give feedback on:
- Do you like the feature?
- Are there better ways to do any of the changes? 
- Am I missing anything to make this change complete? 

I also noticed that the autoformatter that I think was installed by DIM project in VS code seemed to be making a _lot_ of changes to code I didn't edit... Adding newlines, changing indent, adding spaces after `<var>[ ]?.<property>[ ]?.<property>`. Is there a config or setup step I'm missing to stop this? That said, it seems like some of them were reverted as part of the submit process..? So definitely confused.

### CODE REVIEW

**Comment by johyphenel** [comment]
Not sure the right way to "officially" reference bugs? But this relates to #4786.

File: src/data/d2/seasonal-mod-slots.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
+  "none": [],
   "forge": [65589297],
   "opulent": [1962317640, 2712224971, 1202876185],
   "outlaw": [2149155760, 13646368],
-  "undying": [1081029832]
+  "undying": [1081029832],
+  "any": [1081029832, 1202876185, 13646368, 1962317640, 2149155760, 2712224971, 65589297]
 }
```
Comment (delphiactual) [inline src/data/d2/seasonal-mod-slots.json]: You cannot modify this file, you must submit a PR to https://github.com/DestinyItemManager/d2-additional-info to change the way this file is generated.

File: src/app/search/search-filters.ts [hunk 2]
```diff
@@ -1018,11 +1018,20 @@ function searchFilters(
         );
       },
       modslot(item: DimItem, predicate: string) {
-        const modSocketHashes = seasonalSocketHashesByName[predicate];
+        if (!item.isDestiny2() || !item.sockets || !(item.bucket?.sort === 'Armor')) return false;
```
Comment (delphiactual) [inline src/app/search/search-filters.ts]:
```suggestion
        if (!item.isDestiny2() || !item.sockets || !(item.bucket?.sort === 'Armor')) {
          return false;
        }
```

**Comment by delphiactual** [comment]
It seems like you didn't fully follow https://github.com/DestinyItemManager/DIM/blob/master/docs/CONTRIBUTING.md as you were able to submit a PR with bad linting meaning something did not get installed correctly. Are you in our Discord by chance, then we can walk through what is missing for you... most likely you missed `yarn install`

File: src/app/search/search-filters.ts [hunk 3]
```diff
@@ -1018,11 +1018,20 @@ function searchFilters(
...
+                modSocketHashes.includes(socket?.plug?.plugItem?.plug?.plugCategoryHash!)
+              )
+          );
+        }
         return Boolean(
           modSocketHashes &&
-            item.isDestiny2() &&
-            item.sockets &&
             item.sockets.sockets.find((socket) =>
               modSocketHashes.includes(socket?.plug?.plugItem?.plug?.plugCategoryHash)
```
Comment (delphiactual) [inline src/app/search/search-filters.ts]: dedupe code by doing  the following
```
        if (!item.isDestiny2() || !item.sockets || !(item.bucket?.sort === 'Armor')) {
          return false;
        }
        let isNone = false;
        if (predicate === 'none') {
          isNone = true;
          predicate = 'any';
        }
        const modSocketHashes = seasonalSocketHashesByName[predicate];
        const matches = Boolean(
          item.sockets.sockets.find((socket) =>
            modSocketHashes?.includes(socket?.plug?.plugItem?.plug?.plugCategoryHash)
          )
        );
        return !isNone ? matches : !matches;
```

File: src/data/d2/seasonal-mod-slots.json (hunk 1 above)
Comment (delphiactual) [reply src/data/d2/seasonal-mod-slots.json]: https://github.com/DestinyItemManager/d2-additional-info/pull/84

Comment (johyphenel) [reply src/data/d2/seasonal-mod-slots.json]: Thanks for the info, and for drafting the PR :) 

Should we just wait on this PR until DestinyItemManager/d2-additional-info#84 is merged then?

File: src/app/search/search-filters.ts (hunk 2 above)
Comment (johyphenel) [reply src/app/search/search-filters.ts]: Sorry :( fixed.
My work environment auto runs formatters and linters when you create changes, so I have some bad habits :shame-cube:.

File: src/app/search/search-filters.ts (hunk 3 above)
Comment (johyphenel) [reply src/app/search/search-filters.ts]: Good suggestion, thanks! :)

**Comment by johyphenel** [comment]
Updated per comments.
- Reverted manual changes to auto-generated config file. 
- Updated modslot code per suggestion.
- Resorted help file to match autogenerated order in DestinyItemManager/d2-additional-info#84 
- Actually ran `yarn lint` this time :o

File: src/data/d2/seasonal-mod-slots.json (hunk 1 above)
Comment (delphiactual) [reply src/data/d2/seasonal-mod-slots.json]: merging now

**Comment by sundevour** [comment]
works fab for me. thank you!

**Pull request merged**